### ISSUE TITLE
using static properties breaks treeshaking

### ISSUE THREAD

**Issue by montezume** [commercetools/ui-kit#1181]
#### Summary

Take the Card component for example.

If you make a React app that builds with webpack in production mode, and has the following import

```js
import { PinGearIcon } from 'ui-kit';
```
You will end up with ~300-400kb of UI-Kit. Including the "Card" component.

The Card component is a good example / easy one to fix because it is not used by any other component or extended. 

The Card component has three static properties. `propTypes`, `defaultProps` and `displayName`. We use a babel plugin to remove our proptypes, but we use the mode "wrap", which actually does the following.

```js
Card.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  type: PropTypes.oneOf(['raised', 'flat']),
  children: PropTypes.any,
  theme: PropTypes.oneOf(['light', 'dark'])
} : {};
```
As you can see, it is actually assigning an empty object to the propTypes.

I suggest we move to using "unsafe-wrap", which does the following instead.

```js
process.env.NODE_ENV !== "production" ? Card.propTypes = {
  className: PropTypes.string,
  type: PropTypes.oneOf(['raised', 'flat']),
  children: PropTypes.any,
  theme: PropTypes.oneOf(['light', 'dark'])
} : void 0;
```

I made that babel change locally, and treeshaking was still not working.

Then I replaced the defaultProps with this

```js
const Card = ({
  type = 'raised',
  theme = 'light',
  className,
  children,
  ...rest
}) => (
```
Yes, I know @tdeekens does not like destructuring. But it worked.

An alternative is to wrap the displayName in a check for production. Then treeshaking works again. As far as I can tell, we can only use ONE static property at a time.

Read this [thread](https://github.com/alex996/react-css-spinners/issues/1) for more detailed information.

I suggest we kill all usages of `defaultProps` in the library.

**Comment by emmenko** [comment]
> I suggest we kill all usages of `defaultProps` in the library.

Yeah I think we can try that first 💯

**Comment by tdeekens** [comment]
This looks very promising. Thanks for continuously looking into it. To clarify my distaste for destructuring which is not much about this case

1. We have had super (3-4 level) fancy destructuring patterns in our apps which is hard to read makes things super hard to follow as it is unclear where something comes from down the line
2. ESLint was (is not anymore) not able to follow through on nested desturcturing on prop-types giving no warnings if things lacked a prop-type

Having said that for context again. I think what we are doing here is fine. I think defaultProps are actually a bit of uncalled-for syntactic sugar in React which can be emulated in user land. I actually thinkI have actually read considerations of the React team of removing them entirely maybe. 

Lastly, I think, also perfectly fine, emulating defaulProps can not only be done by desturcturing props with defaulting in the argument position of a component. We e.g default options with a merge-options call at times. Which is another pattern to achieve this like `const defaultedProps = defaultProps(props)`. 

Sorry for a bit of text but I wanted to provide the context towards that view.

**Comment by jonnybel** [comment]
I'm also in favour of removing defaultProps.

Especially since React is moving more and more towards components as functions, to me it feels more intuitive to have the defaults defined as function parameters, the ES6 way of doing it, and right at the "head" of the component definition.

And this issue seems a big enough reason to do it.

**Comment by emmenko** [comment]
I think we can close this.
